pkg/client/unversioned/helper.go

```diff
@@ -460,32 +465,25 @@ func DefaultServerURL(host, prefix string, groupVersion unversioned.GroupVersion
 		}
 		hostURL, err = url.Parse(scheme + base)
 		if err != nil {
-			return nil, err
+			return nil, "", err
 		}
 		if hostURL.Path != "" && hostURL.Path != "/" {
-			return nil, fmt.Errorf("host must be a URL or a host:port pair: %q", base)
+			return nil, "", fmt.Errorf("host must be a URL or a host:port pair: %q", base)
 		}
 	}
 
-	// If the user specified a URL without a path component (http://server.com), automatically
-	// append the default prefix
-	if hostURL.Path == "" {
-		if prefix == "" {
-			prefix = "/"
-		}
-		hostURL.Path = prefix
-	}
+	versionedAPIPath := path.Join("/", hostURL.Path, apiPath)
```

Comment:
hostURL.Path contains an optional proxy-prefix. apiPath is generally either
/api or /apis -- it's a pointer to the API entrypoint of the server (where
/api is legacy and /apis is the future). i'd love to get rid of apiPath but
I just don't see how to do that while we need to support both. suggestions?

On Thu, Jan 7, 2016 at 5:36 PM, krousey <notifications@github.com> wrote:

> In pkg/client/unversioned/helper.go
> <https://github.com/kubernetes/kubernetes/pull/17535#discussion_r49136318>
> :
>
> >  		}
> >  	}
> >
> > -	// If the user specified a URL without a path component (http://server.com), automatically
> > -	// append the default prefix
> > -	if hostURL.Path == "" {
> > -		if prefix == "" {
> > -			prefix = "/"
> > -		}
> > -		hostURL.Path = prefix
> > -	}
> > +	versionedAPIPath := path.Join("/", hostURL.Path, apiPath)
>
> This is confusing to me. I would expect hostURL.Path to be empty since it
> should just be the host. What does it mean if host is "
> example.com/prefix_one/" and apiPath is "prefix_two"?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17535/files#r49136318>.
>